aiohttp_client_cache/backends/dynamodb.py

```diff
@@ -1,148 +1,179 @@
-        self.connection = connection or boto3.resource(
-            'dynamodb', region_name=region_name, **kwargs
-        )
-
-        # Create the table if it doesn't already exist
-        try:
-            self.connection.create_table(
-                AttributeDefinitions=[
-                    {
-                        'AttributeName': 'namespace',
-                        'AttributeType': 'S',
-                    },
-                    {
-                        'AttributeName': 'key',
-                        'AttributeType': 'S',
-                    },
-                ],
-                TableName=table_name,
-                KeySchema=[
-                    {'AttributeName': 'namespace', 'KeyType': 'HASH'},
-                    {'AttributeName': 'key', 'KeyType': 'RANGE'},
-                ],
-                ProvisionedThroughput={
-                    'ReadCapacityUnits': read_capacity_units,
-                    'WriteCapacityUnits': write_capacity_units,
-                },
-            )
-        except ClientError:
-            pass
-
-        self._table = self.connection.Table(table_name)
-        self._table.wait_until_exists()
-
-    def _scan_table(self) -> Dict:
-        return self._table.query(
-            ExpressionAttributeValues={':Namespace': self.namespace},
-            ExpressionAttributeNames={'#N': 'namespace'},
-            KeyConditionExpression='#N = :Namespace',
-        )
-
-    @staticmethod
-    def unpickle(response_item: Dict) -> ResponseOrKey:
-        return BaseCache.unpickle((response_item or {}).get('value'))
+        self.key_attr_name = key_attr_name
+        self.val_attr_name = val_attr_name
+        self.create_if_not_exists = create_if_not_exists
+        self.context = context
+        self._table = None
+
+    async def get_table(self):
+        if not self._table:
+            # Re-use the service resource if it's already been created
+            if self.context.cls:
+                conn = self.context.cls
+            # otherwise create
+            else:
+                # should we try to call aexit later if we auto enter here?
+                conn = await self.context.__aenter__()
+
+            self._table = await conn.Table(self.table_name)
+            if self.create_if_not_exists:
+                try:
+                    await conn.create_table(
+                        AttributeDefinitions=[
+                            {
+                                'AttributeName': self.key_attr_name,
+                                'AttributeType': 'S',
+                            },
+                        ],
+                        TableName=self.table_name,
+                        KeySchema=[
+                            {
+                                'AttributeName': self.key_attr_name,
+                                'KeyType': 'HASH',
+                            },
+                        ],
+                        BillingMode="PAY_PER_REQUEST",
```

Comment:
I think the pay-as-you-go is a better default than read/write 1/1 so that it grows with you but definitely don't feel strongly. I think in practice this would really only be used in int tests and I kind of wanted to remove the auto-create feature entirely and leave provisioning as a task for better-suited tools but /shrugs

Reply:
Agreed. Besides tests, I think it's also handy for to have the option to auto-create the table, etc. so a user can quickly try out all the backends using local containers.

This isn't something anyone would be likely to use in production, though. What would really be helpful is including an example CloudFormation template (or Terraform, if you prefer), maybe under the `examples/` folder. I'll leave that up to you, though.